src/plugins/hetero/src/compiled_model.cpp

```diff
@@ -0,0 +1,577 @@
+// #include "perf_counter.hpp"
+#include "plugin.hpp"
+#include "transformations/rt_info/fused_names_attribute.hpp"
+#include "transformations/utils/utils.hpp"
+
+// #include "graph_debug_dump.hpp"
+#include "openvino/op/util/op_types.hpp"
+#include "openvino/util/common_util.hpp"
+#include "ie_algorithm.hpp"
+
+template <typename T>
+using NodeMap = std::unordered_map<ngraph::Node*, T>;
+
+ov::hetero::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
+                                         const std::shared_ptr<const ov::IPlugin>& plugin,
+                                         const Configuration& cfg,
+                                         bool loaded_from_cache)
+    : ov::ICompiledModel(model, plugin),
+      m_cfg(cfg),
+      m_model(model),
+      m_name(model->get_name()),
+      m_loaded_from_cache(loaded_from_cache) {
+    try {
+        bool dumpDotFile = false;
+        if (std::getenv("OPENVINO_HETERO_VISUALIZE")) {
+            dumpDotFile = true;
+        } else {
+            dumpDotFile = m_cfg.dump_graph;
+        }
+
+        ov::SupportedOpsMap queryNetworkResult;
+        auto orderedOps = m_model->get_ordered_ops();
+
+        // TODO vurusovs REMOVE plugin->query_model FROM HERE
+        queryNetworkResult = plugin->query_model(model, {ov::device::priorities(m_cfg.device_priorities)});
+        bool allEmpty = true;
+        // Get user defined affinity
+        for (auto&& node : orderedOps) {
+            auto& nodeInfo = node->get_rt_info();
+            if (node->get_friendly_name() == "conv0")
+                nodeInfo["affinity"] = ov::Any{"GPU"};
+            auto itInfo = nodeInfo.find("affinity");
+            if (itInfo != nodeInfo.end()) {
+                IE_ASSERT(itInfo->second.is<std::string>());
+                queryNetworkResult.emplace(node->get_friendly_name(), itInfo->second.as<std::string>());
+                
+                // TODO vurusovs REMOVE LINE BELOW
+                queryNetworkResult[node->get_friendly_name()] = itInfo->second.as<std::string>();
+                allEmpty = false;
+            }
+        }
+
+        if (queryNetworkResult.empty()) {
+            // here we need to bypass unchanged / unparsed user-set configuration
+            // because it can contain TARGET_FALLBACK / ov::device::priorities
+            queryNetworkResult = plugin->query_model(model, {ov::device::priorities(m_cfg.device_priorities)}); // TODO vurusovs DECIDE ABOUT HETERO m_cfg.GetHeteroConfig()
+        }
+
+        using Input = ov::Input<ov::Node>;
+        using NodeSet = std::unordered_set<ov::Node*>;
+        using InputSet = std::set<Input>;
+
+        auto InputNode = [](const Input& input) {
+            return input.get_source_output().get_node();
+        };
+
+        std::unordered_set<std::string> devices;
+        NodeMap<std::string> affinities;
+        // Check that all nodes has user or plugin defined affinities
+        for (auto&& node : orderedOps) {
```

Comment:
You don't change node context here. It can be:
```suggestion
        for (const auto& node : orderedOps) {
```